### PR TITLE
feat(new server): openttd

### LINKED ISSUE #2454: [Server Request] OpenTTD
## What game server would you like adding?
OpenTTD 

This game is amazingly still in development and offers a wide variety of mods.
Would love to see support for this with LGSM.

https://www.openttd.org/

## Can you link to any tutorials or guides?

https://wiki.openttd.org/Dedicated_server
https://wiki.openttd.org/OpenTTD
https://wiki.openttd.org/Advanced_Settings
https://wiki.openttd.org/Openttd.cfg

## Is the server on Steam?

* [ ] Yes
* [x] No

## Is the server supported on Linux?

* [x] Yes
* [ ] No

### PR DESCRIPTION
**Pull request by ttocszed00**
# Description

Adds support for Open Transport Tycoon Deluxe.
the only thing that doesnt work is update. once they push a new update I can recompile the server.

Fixes #2454

## Type of change

* [ ] Bug fix (a change which fixes an issue).
* [ ] New feature (change which adds functionality).
* [X] New Server (new server added).
* [ ] Refactor (restructures existing code).
* [ ] Comment update (typo, spelling, explanation, examples, etc).

## Checklist

PR will not be merged until all steps are complete.

* [X] This pull request links to an issue.
* [X] This pull request uses the `develop` branch as its base.
* [X] This pull request Subject follows the Conventional Commits standard.
* [X] This code follows the style guidelines of this project.
* [X] I have performed a self-review of my code.
* [X] I have checked that this code is commented where required.
* [X] I have provided a detailed with enough description of this PR.
* [X] I have checked If documentation needs updating.

## Documentation

If documentation does need updating either update it by creating a PR (preferred) or request a documentation update.
* User docs: https://github.com/GameServerManagers/LinuxGSM-Docs
* Dev docs: https://github.com/GameServerManagers/LinuxGSM-Dev-Docs

**Thank you for your Pull Request!**

### CODE REVIEW

**Comment by dgibbs64** [comment]
Ran into a few problems with this.

GLIBC requirements at 2.29 for some reason
No auto-update method

The standard install method is using a deb which fell over when I tried

Maybe a solution is to use the github releases to check version and compile from source?

File: .github/workflows/git-sync.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ name: Backup Repo
 # This will allow LinuxGSM to use Bitbucket as and alternative download if github fails.
 on:
   push:
-    branches:
```
Comment (h3o66) [inline .github/workflows/git-sync.yml:6]: this should not be in the PR

File: lgsm/functions/core_functions.sh [hunk 2]
```diff
@@ -7,7 +7,7 @@
 
 functionselfname="$(basename "$(readlink -f "${BASH_SOURCE[0]}")")"
 
-modulesversion="v20.3.2"
```
Comment (h3o66) [inline lgsm/functions/core_functions.sh:10]: Versions should not be increased.
Please us for this the `clear-functions` of the dev commands for the tests

File: lgsm/functions/install_server_files.sh [hunk 3]
```diff
@@ -52,7 +52,7 @@ fn_install_server_files(){
 	elif [ "${shortname}" == "ut" ]; then
 		remote_fileurl="http://linuxgsm.download/UnrealTournament/UnrealTournament-Server-XAN-3525360-Linux.tar.xz"; local_filedir="${tmpdir}"; local_filename="UnrealTournament-Server-XAN-3525360-Linux.tar.xz";  chmodx="noexecute" run="norun"; force="noforce"; md5="41dd92015713a78211eaccf503b72393"
 	elif [ "${shortname}" == "ut3" ]; then
-		remote_fileurl="http://linuxgsm.download/UnrealTournament3/UT3-linux-server-2.1.tar.xz"; local_filedir="${tmpdir}"; local_filename="UT3-linux-server-2.1.tar.xz"; chmodx="nochmodx" run="norun"; force="noforce"; md5="6b92b083c7ab416399e2183a22fda1df"
+		remote_fileurl="http://linuxgsm.download/UnrealTournament3/UT3-linux-server-2.1.tar.xz"; local_filedir="${tmpdir}"; local_filename="UT3-linux-server-2.1.tar.xz"; chmodx="nochmodx" run="norun"; force="noforce"; md5="8876cca61e3f83ea08db25208bde6ac6"
```
Comment (h3o66) [inline lgsm/functions/install_server_files.sh:55]: For whateven reason this is here

File: linuxgsm.sh [hunk 4]
```diff
@@ -20,7 +20,7 @@ if [ -f ".dev-debug" ]; then
 	set -x
 fi
 
-version="v20.3.2"
```
Comment (h3o66) [inline linuxgsm.sh:23]: Versions should not be increased, see other comment.

File: tests/tests_fctrserver.sh [hunk 5]
```diff
@@ -20,7 +20,7 @@ if [ -f ".dev-debug" ]; then
 	set -x
 fi
 
-version="v20.3.2"
+version="v20.3.3"
```
Comment (h3o66) [inline tests/tests_fctrserver.sh:23]: Versions should not be increased, see other comment.

File: tests/tests_jc2server.sh [hunk 6]
```diff
@@ -20,7 +20,7 @@ if [ -f ".dev-debug" ]; then
 	set -x
 fi
 
-version="v20.3.2"
+version="v20.3.3"
```
Comment (h3o66) [inline tests/tests_jc2server.sh:23]: Versions should not be increased, see other comment.

File: tests/tests_mcserver.sh [hunk 7]
```diff
@@ -20,7 +20,7 @@ if [ -f ".dev-debug" ]; then
 	set -x
 fi
 
-version="v20.3.2"
+version="v20.3.3"
```
Comment (h3o66) [inline tests/tests_mcserver.sh:23]: Versions should not be increased, see other comment.

File: tests/tests_ts3server.sh [hunk 8]
```diff
@@ -20,7 +20,7 @@ if [ -f ".dev-debug" ]; then
 	set -x
 fi
 
-version="v20.3.2"
+version="v20.3.3"
```
Comment (h3o66) [inline tests/tests_ts3server.sh:23]: Versions should not be increased, see other comment.

**Review by h3o66 — CHANGES_REQUESTED** [review]
As well this needs to be rebased to the latest release

**Comment by h3o66** [comment]
> Ran into a few problems with this.
> 
> GLIBC requirements at 2.29 for some reason
> No auto-update method
> 
> The standard install method is using a deb which fell over when I tried
> 
> Maybe a solution is to use the github releases to check version and compile from source?

This should be probably recompiled on a CentOS 7, as this has the lowest version dependency

**Pull request closed without merging**